build.assets/version-check-prerelease/main_test.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestCheck(t *testing.T) {
+	tests := []struct {
+		desc     string
+		tag      string
+		releases []string
+		err      bool
+	}{
+		{
+			desc: "fail-rc",
+			tag:  "v9.0.0-rc.1",
+			err:  true,
+		},
+		{ // this build was published to the deb repos on 2021-10-06
+			desc: "fail-debug",
+			tag:  "v6.2.14-debug.4",
+			err:  true,
+		},
+		{
+			desc: "fail-metadata",
+			tag:  "v8.0.7+1a2b3c4d",
+			err:  true,
+		},
+		{
+			desc: "pass",
+			tag:  "v8.0.1",
+			err:  false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			err := check(test.tag)
+			if test.err && err == nil {
+				t.Fatalf("Expected an error, got nil.")
+			}
+			if !test.err && err != nil {
+				t.Fatalf("Did not expect and error, got: %v", err)
+			}
+		})
+	}
+
```

Comment:
nit:

tests.err -> tests.wantErr (less ambiguous/clearer that it's a bool)
t.Fatalf -> t.Errorf (seems more appropriate to use Error here, and keep going)
```
for _, test := range tests {
    got := check(test.tag)
    if (got != nil) != test.wantErr {
        t.Errorf("check(%s)=%v, want=%t", test.tag, got, test.wantErr)
    }
}
```

Reply:
I made the two prose changes in 7743708.  I chose not to take the entire suggested refatoring as I found the following to be cryptic:
```
--- FAIL: TestCheck (0.00s)                                                                                                                                                                   
    --- FAIL: TestCheck/fail-rc (0.00s)                                                                                                                                                       
        main_test.go:39: check(v9.0.0-rc.1)=<nil>, want=true     
```
vs, the original
```
--- FAIL: TestCheck (0.00s)                                                                                                                                                                   
    --- FAIL: TestCheck/fail-rc (0.00s)                                                                                                                                                       
        main_test.go:39: Expected an error, got nil.
```
I think the couple lines of code savings aren't worth making the test output more confusing.